repos/system_upgrade/el7toel8/actors/postgresqlcheck/tests/test_postgresqlcheck.py

```diff
@@ -0,0 +1,84 @@
+import os
+
+import pytest
+
+from leapp.libraries import stdlib
+from leapp.exceptions import StopActorExecution
+from leapp.libraries.actor import library
+from leapp import reporting
+from leapp.reporting import Report, create_report
+from leapp.libraries.stdlib import api, run
+from leapp.models import RPM, InstalledSignedRPM
+
+RH_PACKAGER = 'Red Hat, Inc. <http://bugzilla.redhat.com/bugzilla>'
+
+
+class produce_mocked(object):
+    def __init__(self):
+        self.called = 0
+        self.model_instances = []
+
+    def __call__(self, *model_instances):
+        self.called += 1
+        self.model_instances.append(model_instances[0])
+
+
+class create_report_mocked(object):
+    def __init__(self):
+        self.called = 0
+        self.report_fields = None
+
+    def __call__(self, **report_fields):
+        self.called += 1
+        self.report_fields = report_fields
+
+
+def test_actor_execution_without_unsigned_data(monkeypatch):
+    def consume_postgresql_packages_mocked(*models):
+        installed_rpm = []
+        yield InstalledSignedRPM(items=installed_rpm)
+    monkeypatch.setattr(api, "consume", consume_postgresql_packages_mocked)
+    monkeypatch.setattr(api, "produce", produce_mocked())
+    monkeypatch.setattr(api, "show_message", lambda x: True)
+    monkeypatch.setattr(reporting, "create_report", create_report_mocked())
+
+    packages = library.get_postgresql_packages()
```

Comment:
I prefer to create the `packages` input directly (if it is valid after possible changes in the actor I mentioned earlier). It doesn't make so much sense to test it in this way. If you want to keep the get_postgresql_packages function, test it separately please.